code/game/machinery/suit_cycler.dm

```diff
@@ -0,0 +1,669 @@
+		return
+
+	if(occupant.client)
+		occupant.client.eye = occupant.client.mob
+		occupant.client.perspective = MOB_PERSPECTIVE
+
+	occupant.forceMove(get_turf(src))
+	occupant = null
+
+	add_fingerprint(usr)
+	updateUsrDialog()
+	update_icon()
+	return
+
+//There HAS to be a less bloated way to do this. TODO: some kind of table/icon name coding? ~Z
+/obj/machinery/suit_cycler/proc/apply_paintjob()
+	if(!target_species || !target_department)
+		return
+
+	if(target_species)
+		if(helmet)
+			helmet.refit_for_species(target_species)
+		if(suit)
+			suit.refit_for_species(target_species)
+
+	switch(target_department)
+		if("Engineering")
+			if(helmet)
+				helmet.name = "engineering voidsuit helmet"
+				helmet.icon_state = "rig0-engineering"
+				helmet.item_state = "eng_helm"
+			if(suit)
+				suit.name = "engineering voidsuit"
+				suit.icon_state = "rig-engineering"
+				suit.item_state = "eng_voidsuit"
+		if("Mining")
+			if(helmet)
+				helmet.name = "mining voidsuit helmet"
+				helmet.icon_state = "rig0-mining"
+				helmet.item_state = "mining_helm"
+			if(suit)
+				suit.name = "mining voidsuit"
+				suit.icon_state = "rig-mining"
+				suit.item_state = "mining_voidsuit"
+		if("Medical")
+			if(helmet)
+				helmet.name = "medical voidsuit helmet"
+				helmet.icon_state = "rig0-medical"
+				helmet.item_state = "medical_helm"
+			if(suit)
+				suit.name = "medical voidsuit"
+				suit.icon_state = "rig-medical"
+				suit.item_state = "medical_voidsuit"
+		if("Security")
+			if(helmet)
+				helmet.name = "security voidsuit helmet"
+				helmet.icon_state = "rig0-sec"
+				helmet.item_state = "sec_helm"
+			if(suit)
+				suit.name = "security voidsuit"
+				suit.icon_state = "rig-sec"
+				suit.item_state = "sec_voidsuit"
+		if("Atmos")
+			if(helmet)
+				helmet.name = "atmospherics voidsuit helmet"
+				helmet.icon_state = "rig0-atmos"
+				helmet.item_state = "atmos_helm"
+			if(suit)
+				suit.name = "atmospherics voidsuit"
+				suit.icon_state = "rig-atmos"
+				suit.item_state = "atmos_voidsuit"
+		if("Captain")
+			if(helmet)
+				helmet.name = "captain voidsuit helmet"
+				helmet.icon_state = "capspace"
+				helmet.item_state = "capspace"
+			if(suit)
+				suit.name = "captain voidsuit"
+				suit.icon_state = "capspace"
+				suit.item_state = "capspace"
+		if("^%###^%$" || "Mercenary")
+			if(helmet)
+				helmet.name = "blood-red voidsuit helmet"
+				helmet.icon_state = "rig0-syndie"
+				helmet.item_state = "syndie_helm"
+			if(suit)
+				suit.name = "blood-red voidsuit"
+				suit.item_state = "syndie_voidsuit"
+				suit.icon_state = "rig-syndie"
+		if("Research")
+			if(helmet)
+				helmet.name = "research voidsuit helmet"
+				helmet.icon_state = "rig0-sci"
+				helmet.item_state = "research_voidsuit_helmet"
+			if(suit)
+				suit.name = "research voidsuit"
+				suit.item_state = "research_voidsuit"
+				suit.icon_state = "rig-sci"
+
+		if("Freelancers")
+			if(helmet)
+				helmet.name = "freelancer voidsuit helmet"
```

Comment:
why are you changing the name if you end up changing it later anyway?

Reply:
it's changing the name here and then adding "refitted" before it later, otherwise you need to copy paste "refitted" everywhere